File: openmrs-client/src/main/java/org/openmrs/mobile/databases/entities/ConceptEntity.java
/*
 * The contents of this file are subject to the OpenMRS Public License
 * Version 1.0 (the "License"); you may not use this file except in
 * compliance with the License. You may obtain a copy of the License at
 * http://license.openmrs.org
 * Software distributed under the License is distributed on an "AS IS"
 * basis, WITHOUT WARRANTY OF ANY KIND, either express or implied. See the
 * License for the specific language governing rights and limitations
 * under the License.
 *
 * Copyright (C) OpenMRS, LLC.  All Rights Reserved.
 */

package org.openmrs.mobile.databases.entities;

import androidx.room.PrimaryKey;
import org.jetbrains.annotations.NotNull;
import org.openmrs.mobile.models.Datatype;
import org.openmrs.mobile.models.Resource;
import androidx.room.ColumnInfo;
import androidx.room.Embedded;
import androidx.room.Entity;
import androidx.room.RoomWarnings;

@Entity(tableName = "concepts")
public class ConceptEntity extends Resource {

    public ConceptEntity() { }

    @Embedded(prefix = "datatype_")
    @SuppressWarnings(RoomWarnings.PRIMARY_KEY_FROM_EMBEDDED_IS_DROPPED)
    private Datatype datatype;

    @ColumnInfo(name = "name")
    private String name;

    public Datatype getDatatype() {
        return datatype;
    }

    public void setDatatype(Datatype datatype) {
        this.datatype = datatype;
    }

    public String getName() {
        return name;
    }

    public void setName(String name) {
        this.name = name;
    }

}

File: openmrs-client/src/main/java/org/openmrs/mobile/databases/entities/ConceptEntity.kt
package org.openmrs.mobile.databases.entities

import androidx.room.ColumnInfo
import androidx.room.Embedded
import androidx.room.Entity
import androidx.room.PrimaryKey
import org.openmrs.mobile.models.Datatype
import org.openmrs.mobile.models.Resource


@Entity(tableName = "concepts")
class ConceptEntity : Resource() {
    @PrimaryKey
    override var id: Long? = null
    @Embedded(prefix = "datatype_")
    var datatype: Datatype? = null
    @ColumnInfo(name = "name")
    var name: String? = null

}
